src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -812,7 +815,12 @@ def sql_query(self) -> str:
     def schema_query(self) -> str:
         """The first operand decide the column attributes of a query with set operations.
         Refer to https://docs.snowflake.com/en/sql-reference/operators-query.html#general-usage-notes"""
-        return self.set_operands[0].selectable.schema_query
```

Comment:
Is this a regression in the sense that the customer did not enable sql simplifier before?

Reply:
Two things happened together to surface this bug we had in `schema_query` generation for set operations with sql simplifier enabled.
1. The old `schema_query` generation for set operations assumed that the overall schema query will be the same as schema query for the first operand.
2. New StringType changes were added where max width of the column will respected by the client.

The users had sql simplifier enabled and they were doing a `DataFrame.union` operation on two string type columns of different max widths. The first operator had a smaller width than the second operator. After the `union`, they ran a `df.cache_result` operation,  which uses the `schema_query` to create a table and insert the result of `union` operation into the table. Since the max width picked here did not allow entry for the second column, the users had a regression.